### ISSUE TITLE
Ontology import should use bulk create/update

### ISSUE THREAD

**Issue by davmlaw** [SACGF/variantgrid#547]
This takes a really long time at the moment.

Should only take a few seconds to pull down all the ontology, then can divide into created/update and do in big batches

**Comment by TheMadBug** [comment]
This has been brought up before, the reason I didn't do it was:
* The need to create stubs if there's a relationship that references stuff that doesn't exit (but should not overwrite anything that does exist)
* The need to create or update if there's proper data for a term.
* Being able to create relationships that referenced terms.
* In the case of PanelApp AU, don't want to pre-cache too much before starting.

Though this should all be pretty easy to work around. The IDs of OntologyTerm are predictable, and could do a quick check at the start to see what already exists in the DB to optimise some behaviour as needed. I can have a go after I've finished handling other tasks.

**Comment by davmlaw** [comment]
For a benchmark, my run today was:

```
python3 manage.py ontology_import --hgnc_sync --mondo_json ${OT_DIR}/mondo.json --hpo_owl ${OT_DIR}/hp.owl --phenotype_to_genes ${OT_DIR}/phenotype_to_genes.txt
```

```
Deleting 2695 stale <class 'ontology.models.models_ontology.OntologyTermRelation'>
This appears stale: OMIM:619092 None
This appears stale: OMIM:619112 None
This appears stale: OMIM:619130 None
Time taken 1:01:47.573622
<class 'ontology.models.models_ontology.OntologyTermRelation'> - Inserts 33028, Updates 373976, Stale 2695
<class 'ontology.models.models_ontology.OntologyTerm'> - Inserts 0, Updates 0, Stale 25
```

**Comment by TheMadBug** [comment]
Ouch, okay, 1 hour. I can def look at that after my current task.

**Comment by davmlaw** [comment]
If you look at import_gene_annotation it does similar things, ie read in everything and then either bulk create or bulk update

**Comment by TheMadBug** [comment]
It's pretty much making it work with the meant to be quicker PanelApp AU live sync that causes the complexity.
Another reason to reconsider that in my opinion, but sure I can work around it.

**Comment by TheMadBug** [comment]
Now bulk imports pre-cache all of ontology, and perform all updates with bulk_create and bulk_update

All imports are now under 3 minutes.
Imports have been run against Shariant Test, and PanelApp AU connection when viewing a gene on the ontology page still works.

**Comment by EmmaTudini** [comment]
@TheMadBug Is there anything that I need to test for this?

**Comment by TheMadBug** [comment]
Nothing specific, just that mondo <-> omim relationships still exist and panel app AU still loads - I've tested both of these things myself.

**Comment by EmmaTudini** [comment]
Testing:
- Logged into test as non-admin assigned to VCGS (emmatudini@gmail.com)
- Selected a new panelapp gene (TNNT2) and searched for gene and "xx" in condition match test
- Imported a variant in a new panelapp gene (NOVA2) - NM_002516.4:c.1A>G (gene symbol autopopulated)

Expected output: TNNT2 and NOVA2 should call API and have condition populated on the HGNC page

Actual output: Passed
Note: Another issue with gene autopopulation and condition matching in #3203

**Comment by EmmaTudini** [comment]
Also imported NM_001330073.1(MPDU1):c.2T>G with condition MPDU1-CDG
Expected output Mondo term should be matched automatically. 
Actual output: Passed

**Comment by EmmaTudini** [comment]
@TheMadBug Any ideas for how I should test MONDO to OMIM relationships? Other than looking at the HGNC page

**Comment by TheMadBug** [comment]
Looking at the HGNC page is the best way to do it.
Alternatively if we look at https://test.shariant.org.au/ontology/term/OMIM_158350 and expand the relationships, we can see that PanelApp AU linked OMIM:158350 to PTEN and that
MONDO:0008021 is exact OMIM:158350

Then if we go to condition matching for PTEN (a few gene symbols down https://test.shariant.org.au/classification/condition_matching/4733) we can see that MONDO:0008021 is suggested as an option due to PanelAPP AU (which it would have gotten by traversing the relationships)
